src/server/cluster/cluster_family.cc

```diff
@@ -607,32 +605,12 @@ void ClusterFamily::DflyClusterFlushSlots(CmdArgList args, ConnectionContext* cn
   return cntx->SendOk();
 }
 
-void ClusterFamily::DflyClusterStartSlotMigration(CmdArgList args, ConnectionContext* cntx) {
-  CmdArgParser parser(args);
-  auto [host_ip, port] = parser.Next<std::string_view, uint16_t>();
-  std::vector<SlotRange> slots;
-  do {
-    auto [slot_start, slot_end] = parser.Next<SlotId, SlotId>();
-    slots.emplace_back(SlotRange{slot_start, slot_end});
-  } while (parser.HasNext());
-
-  if (auto err = parser.Error(); err)
-    return cntx->SendError(err->MakeReply());
-
-  auto* node = AddMigration(std::string(host_ip), port, std::move(slots));
-  if (!node) {
-    return cntx->SendError("Can't start the migration, another one is in progress");
-  }
-  node->Start(cntx);
-
-  return cntx->SendLong(node->GetSyncId());
-}
-
 bool ClusterFamily::StartSlotMigrations(const std::vector<ClusterConfig::MigrationInfo>& migrations,
                                         ConnectionContext* cntx) {
   // Add validating and error processing
   for (auto m : migrations) {
-    auto* node = AddMigration(m.ip, m.port, m.slot_ranges);
+    auto sync_id = CreateOutgoingMigration(m.ip, m.port, m.slot_ranges);
+    auto node = GetOutgoingMigration(sync_id);
```

Comment:
you first create the migration and than get it?
why not change CreateOutgoingMigration to return the created object

also the wording node here is a bit confusing, this is outgoing migration class

Reply:
I need to refactor other places too in this case, I didn't want to create a huge PR

Reply:
please add todo comment